solc.ts

```diff
@@ -104,58 +103,52 @@ function makeSourcePathRelativeIfPossible(sourcePath) {
   for (const absolutePrefix of [absoluteBasePath].concat(absoluteIncludePaths)) {
     const relativeSourcePath = path.relative(absolutePrefix, absoluteSourcePath);
 
-    if (!relativeSourcePath.startsWith('../'))
-      return withUnixPathSeparators(relativeSourcePath);
+    if (!relativeSourcePath.startsWith('../')) { return withUnixPathSeparators(relativeSourcePath); }
   }
 
   // File is not located inside base path or include paths so use its absolute path.
   return withUnixPathSeparators(absoluteSourcePath);
 }
 
-function toFormattedJson(input) {
+function toFormattedJson (input) {
   return JSON.stringify(input, null, program.prettyJson ? 4 : 0);
 }
 
-function reformatJsonIfRequested(inputJson) {
+function reformatJsonIfRequested (inputJson) {
   return (program.prettyJson ? toFormattedJson(JSON.parse(inputJson)) : inputJson);
 }
 
-var callbacks = undefined
-if (options.basePath || !options.standardJson)
-  callbacks = {'import': readFileCallback};
+let callbacks;
+if (options.basePath || !options.standardJson) { callbacks = { import: readFileCallback }; }
 
 if (options.standardJson) {
-  var input = fs.readFileSync(process.stdin.fd).toString('utf8');
-  if (program.verbose)
-    console.log('>>> Compiling:\n' + reformatJsonIfRequested(input) + "\n")
+  const input = fs.readFileSync(process.stdin.fd).toString('utf8');
+  if (program.verbose) { console.log('>>> Compiling:\n' + reformatJsonIfRequested(input) + '\n'); }
   var output = reformatJsonIfRequested(solc.compile(input, callbacks));
 
   try {
-    var inputJSON = smtchecker.handleSMTQueries(JSON.parse(input), JSON.parse(output), smtsolver.smtSolver);
+    const inputJSON = smtchecker.handleSMTQueries(JSON.parse(input), JSON.parse(output), smtsolver.smtSolver);
     if (inputJSON) {
-      if (program.verbose)
-        console.log('>>> Retrying compilation with SMT:\n' + toFormattedJson(inputJSON) + "\n")
+      if (program.verbose) { console.log('>>> Retrying compilation with SMT:\n' + toFormattedJson(inputJSON) + '\n'); }
       output = reformatJsonIfRequested(solc.compile(JSON.stringify(inputJSON), callbacks));
     }
-  }
-  catch (e) {
-    var addError = {
-      component: "general",
+  } catch (e) {
+    const addError = {
+      component: 'general',
       formattedMessage: e.toString(),
       message: e.toString(),
-      type: "Warning"
+      type: 'Warning'
     };
 
-    var outputJSON = JSON.parse(output);
+    const outputJSON = JSON.parse(output);
```

Comment:
How can this be const if it is modified below?

Reply:
Because in Javascript, types are the const not the content.

```
// valid
const example = {} 
example.b = true

// invalid
example = []
```